### PR TITLE
Add support for Linuxbrew.

### PR DESCRIPTION
**Pull request by chenzhekl**
Fix https://github.com/tboox/xmake/issues/295

### CODE REVIEW

File: xmake/modules/lib/detect/pkg_config.lua [hunk 1]
```diff
@@ -85,7 +85,14 @@ function info(name, opt)
...
         -- find the prefix directory of brew directly, because `brew --prefix name` is too slow!
         if not opt.mode or opt.mode == "release" then
-            local pcfile = find_file(name .. ".pc", "/usr/local/Cellar/" .. (opt.brewhint or name) .. "/*/lib/pkgconfig")
+            local brew_pkg_root = try { function () return os.iorunv(brew, {"--prefix"}) end } or "/usr/local"
+            if opt.plat == "macosx" then
+                brew_pkg_root = brew_pkg_root .. "/Cellar/"
+            elseif opt.plat == "linux" then
+                brew_pkg_root = brew_pkg_root .. "/opt/"
+            end
+
```
Comment (waruqi) [inline xmake/modules/lib/detect/pkg_config.lua]: We can simplify line:89-93

```lua
brew_pkg_root = path.join(brew_pkg_root, opt.plat == "macosx" and "Cellar" or "opt")
```

File: xmake/modules/lib/detect/pkg_config.lua [hunk 2]
```diff
@@ -85,7 +85,14 @@ function info(name, opt)
...
+            local pcfile = find_file(name .. ".pc", brew_pkg_root .. (opt.brewhint or name) .. "/*/lib/pkgconfig")
             if pcfile then
```
Comment (waruqi) [inline xmake/modules/lib/detect/pkg_config.lua]: And 

```lua
local pcfile = find_file(name .. ".pc", path.join(brew_pkg_root, (opt.brewhint or name), "*/lib/pkgconfig"))
```

**Comment by chenzhekl** [comment]
Modified per suggestion.

**Comment by waruqi** [comment]
Merged, thanks!

And please commit code to dev branch next time.

**Comment by chenzhekl** [comment]
Sorry, I'll be more careful next time.

**Pull request merged**